### PR TITLE
Added Swift Package Manager

### PR DESCRIPTION
**Pull request by liamrosenfeld**
This PR organizes the files into the standard structure provided by swift.org. The file format should help as we can easily add more targets ([like a CLI](https://github.com/xi-editor/xi-mac/issues/313)). The manifest currently contains all swift targets (so not XiCore) because I'm not sure how to add a compiled rust target as a dependency in the manifest. That means the manifest cannot currently completely replace the project file in the repo (if anyone knows how to, any guidance would be greatly appreciated).

### CODE REVIEW

**Comment by liamrosenfeld** [comment]
Thank you for reviewing the PR. Is there anything left for me to do in order to merge?

**Comment by cmyr** [comment]
@liamrosenfeld cool, thanks for this. I would be curious for feedback from other contributors who have been following recent changes in swift/cocoa land: @terhechte, @lukakerr, @ryanbloom @demalex?

In any case I'll want to give this a look through and think, but it probably makes sense. Has SPM supplanted things like CocoaPods and Carthage, now?

**Comment by cmyr** [comment]
Okay @liamrosenfeld I'm interested in this patch but I don't really feel like I have a good enough sense of what's going on here to make an informed decision. What does SPM buy us? what does it replace? what are its current limitations, etc? If you can point me towards anything that I should read for background info I'd be grateful.

As an aside, we're starting to use zulip as an extra collaborative tool; it might be a helpful way to chat about this? If interested take a look at http://xi.zulipchat.com.

**Comment by liamrosenfeld** [comment]
@cmyr, SPM would immediately provide the project with a more organized directory structure as the manifest enforces the use of 'Sources' and 'Tests' folders. In the medium term, it would act as a dependency manager (which a CLI target would most likely need) and as it's the official dependency manager it would save the headache of switching from Cocoapods/Carthage later on. In the long term, it possibly could replace the Xcode project files. It currently does so in server-side swift, which allows for development in more IDEs then solely Xcode.

Joining the Zulip now.

File: XiEditor.xcodeproj/xcshareddata/xcschemes/XiEditor-Package.xcscheme [hunk 1]
```diff
@@ -0,0 +1,91 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (liamrosenfeld) [reply XiEditor.xcodeproj/xcshareddata/xcschemes/XiEditor-Package.xcscheme]: When I first opened up the Xcode project XiEditor-Package was a scheme. I just assumed that it was involved with something having to do with attaching rust. I'll remove it now.

**Comment by cmyr** [comment]
@liamrosenfeld Okay, took another look through and am happy to get this merged once the conflicts are resolved!

**Comment by liamrosenfeld** [comment]
Resolved all the conflicts!

**Comment by liamrosenfeld** [comment]
Resolved the new conflicts

**Comment by liamrosenfeld** [comment]
I just got a CLI working that is able to open a file inside the editor (working on --wait functionality now).  It's based on this branch so I'll create the PR once this is merged.

**Comment by liamrosenfeld** [comment]
I added Xi-Editor as a submodule because if the Xcode Project is ever replaced it would require a couple more `cd`s during the install process which can get annoying. It is just tracking the master branch instead of any specific version so it should auto update.

**Comment by cmyr** [comment]
@liamrosenfeld Okay one nit here is I'd like the Submodule change to be a separate PR so that we can sort of think about it separately. Ping me after that change and we can merge this. 👍

**Comment by liamrosenfeld** [comment]
@cmyr, I split the submodule commit off.

**Comment by liamrosenfeld** [comment]
Resolved the conflicts.

**Comment by liamrosenfeld** [comment]
Thank you! I moved the submodule commit over to [this PR](https://github.com/xi-editor/xi-mac/pull/331).

**Pull request merged**